electricitymap/contrib/config/__init__.py

```diff
@@ -1,137 +1,62 @@
+(
+    co2eq_parameters_all,
+    co2eq_parameters_direct,
+    co2eq_parameters_lifecycle,
+) = generate_co2eq_parameters(defaults, ZONES_CONFIG)
 CO2EQ_PARAMETERS_DIRECT = {**co2eq_parameters_all, **co2eq_parameters_direct}
 CO2EQ_PARAMETERS_LIFECYCLE = {**co2eq_parameters_all, **co2eq_parameters_lifecycle}
 CO2EQ_PARAMETERS = CO2EQ_PARAMETERS_LIFECYCLE  # Global LCA is the default
 
+# Make a dict mapping each zone to its bounding box.
+ZONE_BOUNDING_BOXES: Dict[ZoneKey, BoundingBox] = zone_bounding_boxes(ZONES_CONFIG)
 
-# Prepare zone bounding boxes
-ZONE_BOUNDING_BOXES: Dict[ZoneKey, BoundingBox] = {}
-for zone_id, zone_config in ZONES_CONFIG.items():
-    if "bounding_box" in zone_config:
-        ZONE_BOUNDING_BOXES[zone_id] = zone_config["bounding_box"]
-
-# Add link from subzone to the full zone
-ZONE_PARENT: Dict[ZoneKey, ZoneKey] = {}
-for zone_id, zone_config in ZONES_CONFIG.items():
-    if "subZoneNames" in zone_config:
-        for sub_zone_id in zone_config["subZoneNames"]:
-            ZONE_PARENT[sub_zone_id] = zone_id
-
-# This object represents the edges of the flow-tracing graph
-def generate_zone_neighbours(
-    zones_config, exchanges_config
-) -> Dict[ZoneKey, List[ZoneKey]]:
-    zone_neighbours = defaultdict(set)
-    for k, v in exchanges_config.items():
-        if not v.get("parsers", {}).get("exchange", None):
-            # Interconnector config has no parser, and will therefore not be part
-            # of the flowtracing graph
-            continue
-        zone_1, zone_2 = k.split("->")
-        if zones_config[zone_1].get("subZoneNames") or zones_config[zone_2].get(
-            "subZoneNames"
-        ):
-            # Both zones must not have subzones
-            continue
-        pairs = [(zone_1, zone_2), (zone_2, zone_1)]
-        for zone_name_1, zone_name_2 in pairs:
-            zone_neighbours[zone_name_1].add(zone_name_2)
-    # Sort
-    return {k: sorted(v) for k, v in zone_neighbours.items()}
-
+# Make a mapping from subzone to the parent zone (full zone).
+ZONE_PARENT: Dict[ZoneKey, ZoneKey] = zone_parents(ZONES_CONFIG)
 
+# Zone neighbours are zones that are connected by exchanges.
 ZONE_NEIGHBOURS: Dict[ZoneKey, List[ZoneKey]] = generate_zone_neighbours(
     ZONES_CONFIG, EXCHANGES_CONFIG
 )
 
-# This object represents all neighbours regardless of granularity
-def generate_all_neighbours(exchanges_config) -> Dict[ZoneKey, List[ZoneKey]]:
-    zone_neighbours = defaultdict(set)
-    for k, v in exchanges_config.items():
-        zone_1, zone_2 = k.split("->")
-        pairs = [(zone_1, zone_2), (zone_2, zone_1)]
-        for zone_name_1, zone_name_2 in pairs:
-            zone_neighbours[zone_name_1].add(zone_name_2)
-    # Sort
-    return {k: sorted(v) for k, v in zone_neighbours.items()}
-
-
 ALL_NEIGHBOURS: Dict[ZoneKey, List[ZoneKey]] = generate_all_neighbours(EXCHANGES_CONFIG)
 
 
-def emission_factors(zone_key: ZoneKey) -> Dict[str, float]:
+def emission_factors(zone_key):
```

Comment:
I kept this here in this change because it's used elsewhere and directly references CO2EQ_PARAMETERS. It could potentially also be moved and refactored to take co2eq_parameters as an argument.